### PR TITLE
Build improvements

### PR DESCRIPTION
**Pull request by emartynov**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description

* Centralise setting of kapt build cache 
* Introduce kapt compile avoidance in the build

## :bulb: Motivation and Context

According to [documentation](https://kotlinlang.org/docs/reference/kapt.html#compile-avoidance-for-kapt-since-1320) it should improve build incremental speed.

## :green_heart: How did you test it?
`./gradlew assembleDebug`

## :pencil: Checklist

- [x] I ran `./gradlew spotlessApply` before submitting the PR
- [x] I reviewed submitted code
- [x] All tests passing

## :crystal_ball: Next steps
Merge and evaluate

## :camera_flash: Screenshots / GIFs

Not the visial change

### CODE REVIEW

**Comment by emartynov** [comment]
Another piece for #718

File: core/build.gradle [hunk 1]
```diff
@@ -86,10 +86,7 @@ dependencies {
     implementation "org.jetbrains.kotlinx:kotlinx-coroutines-android:${versions.coroutines}"
     kapt "com.github.bumptech.glide:compiler:${versions.glide}"
     kapt "com.google.dagger:dagger-compiler:${versions.dagger}"
-}
-
-kapt {
-    useBuildCache = true
+    kapt "androidx.annotation:annotation:${versions.androidx}"
```
Comment (florina-muntenescu) [inline core/build.gradle]: why is this needed?

Comment (emartynov) [reply core/build.gradle]: Because I set up compiler path avoidance I'm getting errors with kapt like:

```
symbol:   class NonNull
  location: class GeneratedAppGlideModuleImpl/ plaid/core/build/generated/source/kapt/debug/com/bumptech/glide/GeneratedAppGlideModuleImpl.java:30: error: cannot find symbol
  public void registerComponents(@NonNull Context context, @NonNull Glide glide,
                                  ^
```

**Comment by emartynov** [comment]
Can someone with admin re-run CI. I can not reproduce the issue locally.

**Comment by keyboardsurfer** [comment]
Your last build passed.

File: core/build.gradle (hunk 1 above)
Comment (emartynov) [reply core/build.gradle]: @florina-muntenescu what are you thoughts?
